src/app/state/stake/stake.effects.spec.ts

```diff
@@ -63,9 +62,13 @@ describe('StakeEffects', () => {
         expect(resultAction).toEqual(PoolActions.initPool());
       });
 
-      effects.initStakingPoolService$.pipe(skip(1)).subscribe(resultAction => {
+      effects.initStakingPoolService$.pipe(skip(1), take(1)).subscribe(resultAction => {
         expect(resultAction).toEqual(PoolActions.getAccountBalance());
       });
+
+      effects.initStakingPoolService$.pipe(skip(2), take(1)).subscribe(resultAction => {
+        expect(resultAction).toEqual(LayoutActions.redirect());
+      });
```

Comment:
I think it could be good to have a comment here as what `skip(2)` vs `skip(1)` represent from a functional standpoint. Why does `skip(1)` mean that you could `getAccountBalance` and `skip(2)` mean that you should `redirect`? Maybe saying that this behaviour "is to fix a race condition between initialization of staking pool and getting list of organizations" would be helpful? I'm not sure if this suggestion below is correct but maybe it gives a starting point?
```suggestion
      /**
       * First we get account balance and then redirect in order to to fix a race condition between initialization of staking pool and getting list of organizations
       */
      effects.initStakingPoolService$.pipe(skip(1), take(1)).subscribe(resultAction => {
        expect(resultAction).toEqual(PoolActions.getAccountBalance());
      });
      effects.initStakingPoolService$.pipe(skip(2), take(1)).subscribe(resultAction => {
        expect(resultAction).toEqual(LayoutActions.redirect());
      });
```

Reply:
account balance is not relevant for the fix. it is needed for staking, but it is called after successful initialization of staking pool. Fix for condition race lays in `redirect` action. So I moved redirect to be called after initialization staking pool instead of successful login like it was previously (race between redirect and initialization staking pool).

Skip part is for skipping previous actions. For example if i want to check third dispatched action, i need to skip two of them. I need to dive into marble testing, because it is easier to read/understand but I couldn't make it work with effects (it's working when i create a custom operator).

However you are right, i will add a comment but in effect file why that redirect is needed.

FYI Dispatching actions is like calling an event (clicking), it's asynchronous and we do not know when it fulfills.